packages/kbn-esql-validation-autocomplete/src/shared/helpers.ts

```diff
@@ -245,7 +253,14 @@ export function lookupColumn(
   { fields, variables }: Pick<ReferenceMaps, 'fields' | 'variables'>
 ): ESQLRealField | ESQLVariable | undefined {
   const columnName = getQuotedColumnName(column);
-  return fields.get(columnName) || variables.get(columnName)?.[0];
+  return (
+    fields.get(columnName) ||
+    variables.get(columnName)?.[0] ||
+    // It's possible columnName has backticks "`fieldName`"
+    // so we need to access the original name as well
+    fields.get(column.name) ||
+    variables.get(column.name)?.[0]
+  );
```

Comment:
So, does this fix a bug?

Reply:
Yes, more specific to this PR. Without it, it yields something like this when adding backticks to the columns. 

<img width="1728" alt="Screenshot 2024-07-30 at 16 42 48" src="https://github.com/user-attachments/assets/2960a1cc-8277-4841-8a77-118ff1ba2ae1">


With those 2 lines:
<img width="1728" alt="Screenshot 2024-07-30 at 16 44 53" src="https://github.com/user-attachments/assets/d5f0c5c2-9cbe-4ccf-b491-8f92d84eba8c">

Reply:
Do the tests cover this case?

Reply:
It's covered by this one https://github.com/elastic/kibana/pull/188600/files#diff-4a3b7269c26dc777be5c0b2a9a1d8f4b1897b7921f6d3e7a176defdf67e83fc6R593